### ISSUE TITLE
Error accessing "models" first time when AnkiDroid not open

### ISSUE THREAD

**Issue by dansanduleac** [ankidroid/Anki-Android#5465]
###### Reproduction Steps

1. Do not use AnkiDroid for a while, or forcibly kill it
2. Call `com.ichi2.anki.api.AddContentApi.getModelList` via its API from another app. The api version used is `com.ichi2.anki:api:1.1.0alpha5`
3. I get an IllegalArgumentException back, "Must set a non-null context to create the configuration."
4. If I try the same action again, it succeeds

###### Expected Result

The `CardContentProvider` is correctly loaded and responds successfully when `getModelList` is called the first time.

###### Actual Result

I get this exception in logcat. This is on 2.9beta14.

```
2019-10-12 21:47:46.263 1625-1645/? I/ActivityManager: Start proc 29971:com.ichi2.anki/u0a233 for content provider com.ichi2.anki/.provider.CardContentProvider
2019-10-12 21:47:46.286 29971-29971/? E/com.ichi2.anki: Not starting debugger since process cannot load the jdwp agent.
2019-10-12 21:47:46.294 1625-17835/? I/ActivityManager: DSS on for com.ichi2.anki and scale is 1.0
2019-10-12 21:47:46.311 29971-29971/? I/com.ichi2.anki: The ClassLoaderContext is a special shared library.
2019-10-12 21:47:46.311 29971-29971/? W/com.ichi2.anki: Insufficient pre-allocated space to mmap vdex.
2019-10-12 21:47:46.330 29971-30003/? E/DatabaseUtils: Writing exception to parcel
    java.lang.IllegalArgumentException: Must set a non-null context to create the configuration.
        at androidx.sqlite.db.SupportSQLiteOpenHelper$Configuration$Builder.build(SupportSQLiteOpenHelper.java:4)
        at com.ichi2.libanki.DB.<init>(DB.java:6)
        at com.ichi2.libanki.Storage.Collection(Storage.java:5)
        at com.ichi2.anki.CollectionHelper.getCol(CollectionHelper.java:6)
        at com.ichi2.anki.provider.CardContentProvider.query(CardContentProvider.java:3)
        at android.content.ContentProvider.query(ContentProvider.java:1078)
        at android.content.ContentProvider.query(ContentProvider.java:1170)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:241)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:102)
        at android.os.Binder.execTransact(Binder.java:739)
```

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [ x ] character to confirm the points below:*

[x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[x] I have searched for similar existing issues here and on the user forum

**Comment by timrae** [comment]
Are you using the latest version of the API?

**Comment by dansanduleac** [comment]
I was using 1.1.0alpha5, but just tested and the same thing happens with
1.1.0alpha6 which I understand to be the latest.

2019年10月13日(日) 3:26 Tim Rae <notifications@github.com>:

**Comment by timrae** [comment]
Well best way is to make a pull request, the idea of the API is just
supposed to be a community developed library to reduce the amount of boiler
plate code in your application.

**Comment by hanpingchinese** [comment]
I'v also just seen this error message out in the wild (crash report) though have not been able to reproduce it yet.  Unfortunately, I don't have any data on which version of AnkiDroid is installed on the user's device (but given the timing of the crash and the release of the latest Ankidroid update, there's a high chance it is the latest stable version). Please can you provide some info on when this error message was added, or maybe which commit might have triggered this?

**Comment by hanpingchinese** [comment]
The exception (in the crash report) is triggered when the client app (my app) calls: 

`resolver.query(Model.CONTENT_URI, null, null, null, null);`

although I have not been able to reproduce this issue.

**Comment by hanpingchinese** [comment]
Okay, so this can be fixed in `AnkiDroidApp.java` by moving: `sInstance = this;` up to just after the `super.onCreate();` call.

**Comment by hanpingchinese** [comment]
Any news on this? It's a super-easy fix. I'm sorry I don't have time to make the PR myself.

**Comment by timrae** [comment]
It's unlikely to be fixed without a PR tbh...

**Comment by david-allison-1** [comment]
Fixed in #5557
